### PR TITLE
coverage - correctly handle root level discriminator

### PR DESCRIPTION
**Pull request by teowa**

### CODE REVIEW

File: coverage/coverage_test.go [hunk 1]
```diff
@@ -44,14 +48,6 @@ func TestCoverage_ResourceGroup(t *testing.T) {
 		t.Fatalf("process coverage: %+v", err)
 	}
 
-	if model.CoveredCount != 1 {
```
Comment (ms-henglu) [inline coverage/coverage_test.go:47]: I guess it's necessary to check whether the model.CoveredCound equals to tc.expectedCoveredCount? And same for the expectedTotalCount?

**Review by ms-henglu — CHANGES_REQUESTED** [review]
Thanks for the PR, LGTM! I only have one comment on the test, please check.

File: coverage/coverage_test.go [hunk 2]
```diff
@@ -1518,29 +1495,37 @@ func testCoverage(t *testing.T, tc testCase) (*coverage.Model, error) {
...
-	//	t.Error(err)
-	//}
-	//t.Logf("coverage model %s", string(out))
+	if model.RootCoveredCount != tc.expectedCoveredCount {
+		t.Errorf("expected CoveredCount %d, got %d", tc.expectedCoveredCount, model.RootCoveredCount)
+	}
+
+	if model.RootTotalCount != tc.expectedTotalCount {
+		t.Errorf("expected TotalCount %d, got %d", tc.expectedTotalCount, model.RootTotalCount)
+	}
```
Comment (teowa) [inline coverage/coverage_test.go:1522]: It's checked in here for every case, and to cope with root level discriminator, `RootTotalCount` and `RootCoveredCount` are added to track the properties count, becase properties count variant of the  root model should NOT be added to the total count of root model

File: coverage/coverage_test.go (hunk 1 above)
Comment (teowa) [reply coverage/coverage_test.go:47]: please see comment below

**Pull request merged**